simde/x86/sse.h

```diff
@@ -2735,12 +2735,24 @@ simde_mm_min_ps (simde__m128 a, simde__m128 b) {
     return _mm_min_ps(a, b);
   #elif defined(SIMDE_FAST_NANS) && defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     return simde__m128_from_neon_f32(vminq_f32(simde__m128_to_neon_f32(a), simde__m128_to_neon_f32(b)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    simde__m128_private
+      r_,
+      a_ = simde__m128_to_private(a),
+      b_ = simde__m128_to_private(b);
+    #if defined(SIMDE_FAST_NANS)
+      r_.wasm_v128 = wasm_f32x4_min(a_.wasm_v128, b_.wasm_v128);
+    #else
+      r_.wasm_v128 = wasm_v128_bitselect(a_.wasm_v128, b_.wasm_v128, wasm_f32x4_lt(a_.wasm_v128, b_.wasm_v128));
+    #endif
+    return simde__m128_from_private(r_);
   #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     simde__m128_private
       r_,
       a_ = simde__m128_to_private(a),
       b_ = simde__m128_to_private(b);
 
+    // we have 2 different implementations because NANs are not valid input for vec_min
```

Comment:
It's not really that NaN is an invalid input, it's just that SSE defines this function a bit oddly.

Comparing NaN to a normal number is always false… i.e., `NAN < 3.14f` and `NAN > 3.14f` are both false. SSE basically treats the min operation as `a < b ? a : b`, so if *either* value is NaN then the result will always be `b`.

For NEON (and most SIMD architectures I'm aware of) min will return NaN (not b) if either value is NaN.